File: python/sort/quick.sort.py
arr = [20,40,30,100,5023,409,48,4348,394]

def partition(arr,low,high):
    pivot = arr[high]
    i = low - 1
    
    for j in range(low,high):
        if arr[j] < pivot:
            i += 1
            arr[i],arr[j] = arr[j],arr[i]
    i += 1
    arr[i],arr[high] = arr[high],arr[i]
    return i

def  quick_sort(arr,low,high):
    if low < high:
        pivot = partition(arr,low,high)
        quick_sort(arr,low,pivot - 1)
        quick_sort(arr,pivot + 1,high)
        
        
print("Original array is: ",arr)

quick_sort(arr,0,len(arr)-1)

print("Sorted array is: ",arr)

File: js/Sort/quick.sort.js
const arr = [1,3,42,50,0,402,305,259,485]


function swap(arr,a,b){
    [arr[a],arr[b]] = [arr[b],arr[a]];
}

function partition(arr,low,high){
    let pivot = arr[high]
    let i = low - 1;
    for(let j = low; j<=high-1; j++){
        if(arr[j] < pivot){
            i++;
            swap(arr,i,j);
        }
    }
    i++;
    swap(arr,i,high)
    return i;
}

function quickSort(arr,low,high){
    if(low < high){
        const pivot = partition(arr,low,high)

        quickSort(arr,pivot + 1,high)
        quickSort(arr,low,pivot - 1)
    }
}


console.log("Original array is: ",arr)

quickSort(arr,0,arr.length-1)


console.log("Sorted array is: ",arr)